packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -306,12 +307,12 @@ class AOTSnapshotter {
     final CompilerOutput compilerOutput = await kernelCompiler.compile(
       sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
       mainPath: mainPath,
-      outputFilePath: fs.path.join(outputPath, 'app.dill'),
+      outputFilePathNoExtension: fs.path.join(outputPath, 'app'),
```

Comment:
I tried to be really explicit about always flowing through output paths without an extension until the final code to write files to avoid bugs where we don't know whether .track should be added.